src/main/scala/uk/gov/hmrc/play/events/AuditEventHandler.scala

```diff
@@ -17,11 +17,12 @@
 package uk.gov.hmrc.play.events
 
 import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import uk.gov.hmrc.play.audit.http.HeaderCarrier
 import uk.gov.hmrc.play.audit.http.connector.AuditConnector
 
 import scala.concurrent.Future
 
-trait AuditEventHandler extends EventHandler {
+abstract class AuditEventHandler(implicit headerCarrier: HeaderCarrier) extends EventHandler {
```

Comment:
Yes, the auditConnector interface used here always requires a HeaderCarrier. If you changed it, you'd have to write your own AuditEventHandler anyway.